articles/per_package_documentation.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: default
+title: Per-Package Documentation
+permalink: articles/per_package_documentation.html
+abstract: This article describes the requirements and design of ROS 2’s per-package documentation system.
+author: Marya Belanger
+published: true
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Background
+
+ROS 2 is lacking a process for the conglomeration of package documentation.
+Discoverability for package documentation (and documentation in general) is one of the most frequent user complaints.
+
+The goal of this design is to standardize a documentation process for ROS 2 packages to encourage maintainers to robustly document their packages.
+The results will be presented in a consistent way alongside the general ROS 2 documentation, making the documentation as a whole discoverable and consistent.
+
+In general, our vision for the system is:
+
+- Package maintainers document their packages in their repositories following some guidelines or templates we recommend
+- Package documentation from the repositories is built on the ROS 2 infrastructure and deployed to the ROS 2 documentation site in an automatic process with no input from the maintainer besides an opt-in `doc` block in the distribution.yaml
```

Comment:
"with no input from the maintainer besides an opt-in `doc` block", I'm not sure what this fragment is trying to get across. Of course the maintainer has some input, they would have needed to merge the PRs with the docs into the branch to build on the build farm. 

I think maybe what you mean isn't "input" but "action" but either way, I think the "deployed to the ROS 2 documentation site in an automatic process" makes that clear.

Reply:
We wanted to emphasize the limited amount of investment on the maintainers' end, but you're right it's not "no input". 
Making sure to incorporate @sloretz's point about [how one would opt in](https://github.com/ros2/design/pull/301#discussion_r499733468), I think the following satisfies your issue?

> Package documentation from the repositories is built on the ROS 2 infrastructure and deployed to the ROS 2 documentation site in an automatic process maintainers can opt in to by adding a `doc` block to the `distribution.yaml`